test/blackbox/api/dds-pim/PubSubTypeTraits.hpp

```diff
@@ -0,0 +1,102 @@
+// Copyright 2024 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file PubSubTypeTraits.hpp
+ *
+ */
+
+#include <fastdds/dds/core/LoanableSequence.hpp>
+#include <fastdds/dds/topic/TypeSupport.hpp>
+
+#include "../../common/BlackboxTests.hpp"
+
+#ifndef _TEST_BLACKBOX_PUBSUBTYPETRAITS_HPP_
+#define _TEST_BLACKBOX_PUBSUBTYPETRAITS_HPP_
+
+/**
+ * @brief This class provides TypeSupport specific traits for the PubSub wrapper classes
```

Comment:
Would be nice if we could have the traits on `PubSubWriterReader` and `PubSubParticipant`